### PR TITLE
Bump borsh to 0.8.0

### PR DESCRIPTION
**Pull request by mikedotexe**
Frol pushed an update to Borsh shown here:
https://crates.io/crates/borsh/0.8.0

Additional notes from Frol:

>Please, release near-sdk-rs with bumped borsh version to avoid compilation errors that are caused by our usage of "private" APIs of syn crate

The second commit contains `Cargo.lock` updates as well as changes for the `status-message` example.
Some of these were artifacts after locally running the `./ci-test.sh` file.

### CODE REVIEW

**Review by frol — APPROVED** [review]
I  am giving my approval to unblock this PR in case all the comments are resolved and it is ready to be merged and released

File: examples/status-message/Cargo.lock [hunk 1]
```diff
@@ -45,7 +51,17 @@ version = "0.7.1"
 source = "registry+https://github.com/rust-lang/crates.io-index"
 checksum = "30f3fd65922359a7c6e791bc9b2bba1b977ea0c0b96a528ac48007f535fb4184"
 dependencies = [
- "borsh-derive",
+ "borsh-derive 0.7.1",
```
[the remark this answers is not in the document]
Comment (mikedotexe) [reply examples/status-message/Cargo.lock:54]: I saw this in a few places when reviewing. Some seemed to be from tags. Looks like this one isn't, but it's unclear to me the best way to proceed for `borsh-derive`.

**Comment by mikedotexe** [comment]
Is it normal for the tests to fail for this repo? Seems to be happening for:
`NEAR_RELEASE=false`

File: near-sdk/Cargo.toml [hunk 2]
```diff
@@ -21,7 +21,7 @@ path = "compilation_tests/all.rs"
 serde = { version = "1", features = ["derive"] }
 serde_json = "1"
 near-sdk-macros = { path = "../near-sdk-macros", version = "2.0.0"}
-borsh = "0.7.0"
+borsh = "0.8.0"
```
[the remark this answers is not in the document]
Comment (mikedotexe) [reply near-sdk/Cargo.toml:24]: I'm going to have to ask @evgenykuzyakov to help this time and pay attention to how this works so I can answer in the future.

File: examples/status-message/Cargo.lock (hunk 1 above)
[the remark this answers is not in the document]
Comment (frol) [reply examples/status-message/Cargo.lock:54]: Try running `cargo update` in the folder.

**Comment by birchmd** [comment]
Dropping a line on this PR since it doesn't appear to have had any activity for several days. What is the status on it? It would be nice to get this merged because the compilation issue is preventing nightly runs of the mocknet load test (see e.g. https://nayduckdiag.blob.core.windows.net/logs/75542_v2_stderr ). There is also at least one person in the community who is experiencing friction because of the compilation issue as well: https://discordapp.com/channels/490367152054992913/714898996832895027/803300211711279174

**Comment by evgenykuzyakov** [comment]
Hey, sorry. The borsh was not a blocker. It's `syn` issue. I'll rebase my PR that I used to release `2.0.1` https://github.com/near/near-sdk-rs/pull/266 and merge it to `master`. This should fix `master` and dependencies.

**Comment by birchmd** [comment]
@evgenykuzyakov Unfortunately pinning the `syn` version for `near-sdk-core` seems to not be enough. I think it is still important to have it depend on the new borsh version. The [compilation still fails in nightly testing](https://nayduckdiag.blob.core.windows.net/logs/76343_v2_stderr). You can see in the output that it is still compiling `syn 1.0.60` (a later version from the pinned one). I looked at the `Cargo.lock` to see where the various dependencies are coming from and see the following:

```
[[package]]
name = "borsh-derive"
version = "0.7.2"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "e6aaa45f8eec26e4bf71e7e5492cf53a91591af8f871f422d550e7cc43f6b927"
dependencies = [
 "borsh-derive-internal",
 "borsh-schema-derive-internal",
 "proc-macro2 1.0.24",
 "syn 1.0.60",
]

[[package]]
name = "borsh-derive-internal"
version = "0.7.2"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "61621b9d3cca65cc54e2583db84ef912d59ae60d2f04ba61bc0d7fc57556bda2"
dependencies = [
 "proc-macro2 1.0.24",
 "quote 1.0.8",
 "syn 1.0.60",
]

[[package]]
name = "borsh-schema-derive-internal"
version = "0.7.2"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "85b38abfda570837b0949c2c7ebd31417e15607861c23eacb2f668c69f6f3bf7"
dependencies = [
 "proc-macro2 1.0.24",
 "quote 1.0.8",
 "syn 1.0.60",
]
...
[[package]]
name = "near-rpc-error-core"
version = "0.1.0"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "ffa8dbf8437a28ac40fcb85859ab0d0b8385013935b000c7a51ae79631dd74d9"
dependencies = [
 "proc-macro2 1.0.24",
 "quote 1.0.8",
 "serde",
 "serde_json",
 "syn 1.0.60",
]

[[package]]
name = "near-rpc-error-macro"
version = "0.1.0"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "0c6111d713e90c7c551dee937f4a06cb9ea2672243455a4454cc7566387ba2d9"
dependencies = [
 "near-rpc-error-core",
 "proc-macro2 1.0.24",
 "quote 1.0.8",
 "serde",
 "serde_json",
 "syn 1.0.60",
]
```

The borsh and near-rpc-error dependencies still pull in `syn 1.0.60`. Later in the file we get to the `near-sdk-core` dependencies:

```
[[package]]
name = "near-sdk-core"
version = "2.0.1"
source = "git+https://github.com/near/near-sdk-rs?branch=master#bf402b6a4ccf09c221ddb45c4993eb15c37bc847"
dependencies = [
 "Inflector",
 "proc-macro2 1.0.24",
 "quote 1.0.8",
 "syn 0.15.44",
]

[[package]]
name = "near-sdk-macros"
version = "2.0.1"
source = "git+https://github.com/near/near-sdk-rs?branch=master#bf402b6a4ccf09c221ddb45c4993eb15c37bc847"
dependencies = [
 "near-sdk-core",
 "proc-macro2 1.0.24",
 "quote 1.0.8",
 "syn 0.15.44",
]
```

You can see they bring in an older version of `syn`. Notably they bring in a pre-1.0 version, which seems suspicious to me since I thought the version we wanted everything to depend on was `1.0.57`.

The `Cargo.toml` file used for this compilation is the one from [near-core pytests](https://github.com/near/nearcore/blob/9becd3cb52e912f7eeda55b3369f8bf9ed6e5645/pytest/empty-contract-rs/Cargo.toml). You can see its only dependency is the master branch of `near-sdk-rs`, so this compilation issue is something I think should be fixed in this repo.

I did also try specifying the `syn` version in that `Cargo.toml` hoping it could be a work around, but it still would not compile.

Any further help you could offer with this would be appreciated 🙏

**Comment by evgenykuzyakov** [comment]
Closing. Fixed as part of new release

**Pull request closed without merging**